spyder/plugins/editor/widgets/tests/test_editor_and_outline.py

```diff
@@ -35,16 +35,16 @@ def test_files(tmpdir_factory):
                 "def foo:\n"
                 "    print(Hello World!)\n")
 
-    filename2 = osp.join(tmpdir.strpath, 'foo2.py')
+    filename2 = osp.join(tmpdir.strpath, 'text1.txt')
     with open(filename2, 'w') as f:
-        f.write("# -*- coding: utf-8 -*-\n"
-                "# ---- a comment\n")
-
-    filename3 = osp.join(tmpdir.strpath, 'text1.txt')
-    with open(filename3, 'w') as f:
         f.write("This is a simple text file for\n"
                 "testing the Outline Explorer.\n")
 
+    filename3 = osp.join(tmpdir.strpath, 'foo2.py')
+    with open(filename3, 'w') as f:
```

Comment:
You should add ``encoding="utf-8" for consistency across platforms and general standard conformance (since on Windows this will default to the locale encoding, which could theoretically be anything and may or may not read the file correctly or completly), especially since you define the file encoding as UTF-8 below.

Reply:
This is not Python 2.7 friendly it seems...

Reply:
... so reverting since everything is ASCII in those test files anyway.

Reply:
Ah. Looking it up, apparently the best practice is to just always use ``io.open()`` instead of open() if needed Python 2 compatible code, as it includes an encoding argument and works in Python 2 and Python 3 identically; in fact, in Python 3, open() is just an alias for io.open(). This ensures compatibility between the two versions and ensures files will be decoded and encoded consistently and with full Unicode compatibility.